src/game/client/worldheightmap.cpp

```diff
@@ -29,21 +45,2358 @@ class GDIFileStream : public InputStream
+        i2 = width;
+    }
+
+    for (int i = 0; i < width; i++) {
+        memcpy(&buffer[4 * width * i], &rgb[8 * width * (i2 + i) + 4 * i1], 4 * width);
+    }
+
+    return true;
+}
+
+unsigned char *WorldHeightMap::Get_Pointer_To_Tile_Data(int x, int y, int width)
+{
+    int i1 = x + m_width * y;
+
+    if (y < 0 || x < 0 || x >= m_width || y >= m_height) {
+        return nullptr;
+    }
+
+    if (i1 < 0 || i1 >= m_dataSize) {
+        return nullptr;
+    }
+
+    int i2 = m_blendTileNdxes[i1];
+
+    if (i2 > 0 && i2 < 16192 && Get_Raw_Tile_Data(m_blendedTiles[i2].blend_ndx, width, s_blendBuffer, 16384)) {
+        unsigned char *alpha = Get_RGB_Alpha_Data_For_Width(width, &m_blendedTiles[i2]) + 3;
+        unsigned char *blendbuf = s_blendBuffer;
+        unsigned char *buf = s_buffer;
+
+        for (int i = 0; i < width * width; i++) {
+            int i3 = *blendbuf;
+            unsigned char *blendbuf2 = blendbuf + 1;
+            int i4 = *blendbuf2++;
+            int i5 = *blendbuf2;
+            blendbuf = blendbuf2 + 2;
+            int i6 = *alpha;
+            alpha += 4;
+            *buf = (255 - i6) * *buf / 255 + i6 * i3 / 255;
+            unsigned char *buf2 = buf + 1;
+            *buf2 = (255 - i6) * *buf2 / 255 + i6 * i4 / 255;
+            buf2++;
+            *buf2 = (255 - i6) * *buf2 / 255 + i6 * i5 / 255;
+            buf2[1] = 255;
+            buf = buf2 + 2;
+        }
+    }
+
+    return s_buffer;
+}
+
+unsigned char *WorldHeightMap::Get_RGB_Alpha_Data_For_Width(int width, TBlendTileInfo *info)
+{
+    int i1 = 0;
+
+    if (info->horiz) {
+        i1 = 0;
+    } else if (info->vert) {
+        i1 = 1;
+    } else if (info->right_diagonal) {
+        i1 = 3;
+        if (info->long_diagonal) {
+            i1 = 5;
+        }
+    } else if (info->left_diagonal) {
+        i1 = 2;
+        if (info->long_diagonal) {
+            i1 = 4;
+        }
+    }
+
+    if (info->inverted) {
+        i1 += 6;
+    }
+
+    return s_alphaTiles[i1]->Get_RGB_Data_For_Width(width);
+}
+
+void WorldHeightMap::Clear_Seismic_Update_Flags()
+{
+    if (m_seismicUpdateFlags) {
+        memset(m_seismicUpdateFlags, 0, m_height * m_siesmicUpdateWidth);
+    }
+}
+
+void WorldHeightMap::Fill_Seismic_Z_Velocities(float value)
+{
+    if (m_siesmicZVelocity) {
+        for (int i = 0; i < m_height * m_width; i++) {
+            m_siesmicZVelocity[i] = value;
+        }
+    }
+}
+
+void WorldHeightMap::Init_Cliff_Flags_From_Heights()
+{
+    for (int w = 0; w < m_width - 1; ++w) {
+        for (int h = 0; h < m_height - 1; ++h) {
+            Set_Cell_Cliff_Flag_From_Heights(w, h);
+        }
+    }
+}
+
+int WorldHeightMap::Get_Texture_Class_From_Ndx(int tile_ndx)
+{
+    int i1 = tile_ndx >> 2;
+
+    for (int i = 0; i < m_numTextureClasses; ++i) {
+        if (m_textureClasses[i].first_tile >= 0 && i1 >= m_textureClasses[i].first_tile
+            && i1 < m_textureClasses[i].num_tiles + m_textureClasses[i].first_tile) {

```

Comment:
asm shows `m_textureClasses[i].first_tile + m_textureClasses[i].num_tiles` which is more logical

Reply:
Fixed everywhere I can see doing this.